lib/bunch/sref.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Bunch.Sref do
+  @moduledoc """
+  A bunch of stuff for shortly-inspectable references.
+  """
+  @enforce_keys [:ref, :hash]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{ref: reference, hash: String.t()}
+
+  @doc """
+  Creates a shortly-inspectable reference.
+
+      iex> IEx.Helpers.ref(0, 1, 2, 3) |> #{inspect(__MODULE__)}.new() |> inspect()
+      "#6f1ef15f"
+      iex> <<"#", hash::binary-size(8)>> = #{inspect(__MODULE__)}.new() |> inspect()
+      iex> Base.decode16(hash, case: :lower) |> elem(0)
+      :ok
+
+  """
+  @spec new(reference) :: t
+  def new(ref \\ make_ref()) do
+    '#Ref' ++ ref_list = :erlang.ref_to_list(ref)
```

Comment:
If you only calculate hash - what's the point of splitting this charlist?

Reply:
it reduces the likelihood of a conflict

Reply:
That's not how hash works :P Similarity of input data doesn't imply similarity of the output.
If it did, that wouldn't be a good hash.
It is even quite common when storing hashed passwords to add a common (and secret) part called 'pepper' https://en.wikipedia.org/wiki/Pepper_(cryptography)

What's more:
```
iex(7)> make_ref |> :erlang.ref_to_list
'#Ref<0.2562459300.1782317058.164570>'
iex(8)> make_ref |> :erlang.ref_to_list
'#Ref<0.2562459300.1782317058.164577>'
iex(9)> make_ref |> :erlang.ref_to_list
'#Ref<0.2562459300.1782317060.164483>'
iex(10)> make_ref |> :erlang.ref_to_list
'#Ref<0.2562459300.1782317060.164490>'
iex(11)> make_ref |> :erlang.ref_to_list
'#Ref<0.2562459300.1782317060.164497>'
```
If what you said was true we'd had much bigger problem than `'#Ref'`

Reply:
> That's not how hash works :P Similarity of input data doesn't imply similarity of the output.

It's not about the similarity, it's about length. The longer the input, the bigger the space of possible inputs, the bigger chance of a conflict.

> It is even quite common when storing hashed passwords to add a common (and secret) part called 'pepper' 

'(and secret)' is the key there ;) The use of hash doesn't have anything to do with security in this case.

Reply:
> It's not about the similarity, it's about length. The longer the input, the bigger the space of possible inputs, the bigger chance of a conflict.

You're missing an important detail - if the prefix is constant, the space size of inputs doesn't change at all.
What's more, SHA works on blocks of 512-bits, so the input will be padded with 0s (up to 448 bits) and that makes stripping the prefix a complete waste of time.

Reply:
It does grow if you don't know the prefix is constant, as SHA algorithm doesn't. However, if it works on blocks as you said, I agree